supporter-product-data/src/main/scala/com/gu/lambdas/AddSupporterRatePlanItemToQueueLambda.scala

```diff
@@ -98,60 +105,17 @@ object UpdateDynamoLambda extends StrictLogging {
       logger.info(
         s"Continuing processing with batch of ${batch.length} - time remaining: ${timeOutCheck.timeRemainingMillis / 1000} seconds, buffer: ${timeoutBufferInMillis / 1000} seconds",
       )
-
-      Await.result(writeBatch(batch, dynamoDBService, alarmService), 120.seconds)
+      try {
+        sqsService.sendBatch(batch)
+      } catch {
+        case e: Throwable =>
+          SafeLogger.error(scrub"Error sending batch to sqs\nbatch content was ${batch.asJson.spaces2}", e)
```

Comment:
what are we going to do if there is some issue that makes us burn through all the batches from zuora and fail?  Are we going to have a log full of batches and no easy way to get back again?
I'm just wondering if there's a way to put the batch progress so far into the state and retry the batch until it works or it's manually recovered.

Reply:
We could throw at this point  and it will just keep retrying but I think I'd rather trigger an alarm and keep going for now. We can always run a full sync at any time so we're not losing any data permanently.